eras/alonzo/impl/testlib/Test/Cardano/Ledger/Alonzo/Imp/UtxoSpec.hs

```diff
@@ -52,14 +55,23 @@ spec = describe "UTXO" $ do
           submitFailingTx tx [injectFailure $ ExUnitsTooBigUTxO maxExUnits txExUnits]
 
         it "Insufficient collateral" $ do
-          scriptInput <- produceScript . hashPlutusScript $ alwaysSucceedsWithDatum slang
-          let collateral = Coin 123
+          scriptInput <- produceScript $ hashPlutusScript $ alwaysSucceedsWithDatum slang
           collateralAddr <- freshKeyAddr_
-          collateralInput <- sendCoinTo collateralAddr collateral
-          let tx =
+          collateralInput <- sendCoinTo collateralAddr mempty -- 0 will be changed to MinUTxO
+          collateral <- (^. coinTxOutL) <$> impLookupUTxO collateralInput
+          -- We need to artificially blow up the fee to increase the required collateral.
+          -- Unfortunately we do not have expensive enough scripts yet, so one other way
+          -- to achieve the same thing is by increasing the size of the transactions by
+          -- including random garbage. Auxiliary data fits the bill quite nicely
+          metadata <-
+            Map.fromList
+              <$> forM [1 .. (12 * 1024 `div` 64)] (\ix -> (,) ix . M.B <$> uniformByteStringM 64)
+          let auxData = mkAlonzoTxAuxData @[] @era metadata []
+              tx =
                 mkBasicTx mkBasicTxBody
                   & bodyTxL . inputsTxBodyL <>~ [scriptInput]
                   & bodyTxL . collateralInputsTxBodyL <>~ [collateralInput]
+                  & auxDataTxL .~ SJust auxData
```

Comment:
This causes `MissingTxBodyMetadataHash` in `UtxowSpec` to fail. 
But you can fix it easily by adding these  two lines before [submitFailingTx](https://github.com/IntersectMBO/cardano-ledger/blob/lehins/various-fixes-to-imptests/eras/shelley/impl/testlib/Test/Cardano/Ledger/Shelley/Imp/UtxowSpec.hs#L92): 

```
    let resetAddrWits = updateAddrTxWits . (witsTxL . addrTxWitsL .~ [])
    withPostFixup (resetAddrWits . (bodyTxL . auxDataHashTxBodyL .~ SNothing )) $
```

Reply:
I was confused for a second how this test could be causing another test to fail, but looking further into the test failure I realized that you meant the `fixupAuxDataHash` that was needed for this test is causing the `MissingTxBodyMetadataHash` to fail.

Unfortunately, your suggestion will not work, since removing witnesses will cause the transaction to fail, cause signatures for the inputs being spent will be missing. We can't change just the `auxDataHashTxBodyL` after fixup either, because it will affect signatures.

A much easier fix to this is to add the actual `auxData` after the fixup :slightly_smiling_face: 

```haskell
        tx = mkBasicTx $ mkBasicTxBody & feeTxBodyL .~ Coin 165545
        addAuxData fixedTx = pure (fixedTx & auxDataTxL .~ SJust auxData)
    withPostFixup addAuxData $
```

One caveat is that the fee must be set manually, since addition of aux data affects the size of the transaction consequently making the fixud up fee insufficient. I've also added functionality that prevents mocking with the `feeTxBodyL` field if it is already present.